azurelinuxagent/common/telemetryevent.py

```diff
@@ -39,6 +39,15 @@ def __init__(self, eventId=None, providerId=None):
     def __contains__(self, param_name):
         return param_name in [param.name for param in self.parameters]
 
+    def is_extension_event(self):
+        # Events originating from the agent have "WALinuxAgent" as the Name parameter, or they don't have a Name
+        # parameter, in the case of log and metric events. So, in case the Name parameter exists and it is not
+        # "WALinuxAgent", it is an extension event.
+        for param in self.parameters:
+            if param.name == "Name":
```

Comment:
Can there be any extension event that doesn't have any `Name` parameter at all either? (Not sure if its possible for them to generate such events either)

Just a thought, since we can control the events being generated by the agent, can't we add a flag like `agent_event=True` on every event generated by us and use that to recognize if its ours or not? We can remove that flag before sending out the event to wireserver to not pollute wireserver.

Reply:
An extension event without the `Name` parameter already populated wouldn't be a valid event, so we don't care about "messing" them up anyway. I find this on-the-fly logic easier to maintain, having a flag introduces the possibility of human error (forgetting to set the flag in a new agent method, for example).

Reply:
I share Laveesh's concern about extensions forgetting to set the Name parameter. If not a flag, let's try to find another way to handle this.